shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/usecase/GetSurveysUseCase.kt

```diff
@@ -5,14 +5,18 @@ import co.nimblehq.ic.kmm.suv.domain.repository.SurveyRepository
 import kotlinx.coroutines.flow.Flow
 
 interface GetSurveysUseCase {
-    operator fun invoke(pageNumber: Int, pageSize: Int): Flow<List<Survey>>
+    operator fun invoke(pageNumber: Int, pageSize: Int, isRefresh: Boolean): Flow<List<Survey>>
 }
 
 class GetSurveysUseCaseImpl(
     private val surveyRepository: SurveyRepository
 ) : GetSurveysUseCase {
 
-    override operator fun invoke(pageNumber: Int, pageSize: Int): Flow<List<Survey>> {
-        return surveyRepository.getSurveys(pageNumber, pageSize)
+    override operator fun invoke(
+        pageNumber: Int,
+        pageSize: Int,
+        isRefresh: Boolean
+    ): Flow<List<Survey>> {
+        return surveyRepository.getSurveys(pageNumber, pageSize, isRefresh)
```

Comment:
@luongvo IMO, we should not. When the user open the app again (the first page number is 1), we should fetch the local data, instead of resetting the local data and getting again data from API.

Reply:
@suho Actually, I'm not asking to reset the local data first. We can fetch the local data, fetch the remote data, and then only clear the local data if it's the first page & remote data fetching is successful 💭 Does it make sense?

I'm wondering what happens if we call this API with page = 2 and isRefresh = true?

Reply:
>I'm wondering what happens if we call this API with page = 2 and isRefresh = true? What should we expect in unit testing?

@luongvo Technically, when we want to refresh a list of data, we will need to clear all the data and then fetch the first page. Calling the second page with isRefresh = true does not make sense to me.

> We can fetch the local data, fetch the remote data, and then only clear the local data if it's the first page & remote data fetching is successful
I think clearing the cache only when the pull-to-refresh is the requirement (just checked again [on Notion](https://www.notion.so/nimblehq/Internal-Certification-Path-Mobile-de354e1ce76f4b80922c3db21ea0117f#b6cc9938ec2e419394d4dd8806c154be), so I think it's good to go 🙏

Reply:
> Calling the second page with isRefresh = true does not make sense to me.

@suho yes, but that's what the actual implementation is because the VM can call it with `page != 1` and `isRefresh = true`. I meant, we're leaving the responsibility for the VM instead of controlling that logic at the repository level. 

> I'm wondering what happens if we call this API with page = 2 and isRefresh = true? What should we expect in unit testing?

Exposing an API with parameters at the repository level requires us to cover all possible cases with it by unit testing. In this case, I don't really know what should we do with it.

From the system design perspective, it's not a good design I think. Just my thought, if you think it's ok to go, I can skip this 👌

Reply:
@luongvo 
> I meant, we're leaving the responsibility for the VM instead of controlling that logic at the repository level.

> Exposing an API with parameters at the repository level requires us to cover all possible cases with it by unit testing. In this case, I don't really know what should we do with it.

IMO, we are not always using all parameters to test a case, right? 🤔
For example:
```kt
@Test
fun `when isRefresh is true, it returns the first page of surveys lits`() {...}
```
With this test, do we really need to care about the page number? 🤔

Moreover, just curious if we have this function:
```kt
fun doSomething(shouldUseDefault: Boolean, defaultValue: String) : String {
	if shouldUseDefault {
		return defaultValue
	else {
		return "hello world"
	}	
}
```
If we want to write a unit test for that, if `shouldUseDefault == false`, then we should not care about the default value, am I correct? 🤔